### ISSUE TITLE
NER MoreData_ShouldNotDegrade asserts per-step monotonicity, which SGD does not have — LegalBERTNER fails it nightly

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#2135]
## Summary

`NERModelTestBase.MoreData_ShouldNotDegrade` compares a 1-step run against a 2-step run and requires the longer one to be no worse. That is per-step monotonicity, which stochastic optimisation does not provide — and `NeuralNetworkModelTestBase` was already migrated away from exactly this shape, for exactly this reason.

`LegalBERTNER` fails it in the nightly `Heavy Timeout Models` lane:

```
2-iteration clone loss (3.992393) > 1-iteration loss (2.825676)
  — measured via GetLastLoss (the model's own CE over logits).
    Optimizer may be diverging with more training.
```

Found while investigating #2087; it was hidden behind that lane's runner OOM, so the lane will still be red after #2134 fixes the OOM.

## The base class already solved this

`NeuralNetworkModelTestBase.MoreData_ShouldNotDegrade` measures an **untrained** baseline and asserts the model beats it after one adequate budget. Its comment is explicit about why short-vs-long does not work:

> Comparing a short run against a longer one asserts that loss falls monotonically between two arbitrary iteration counts, which is not a property stochastic gradient descent has: with momentum or Adam the first few steps routinely overshoot before settling. The optical flow family made that concrete — measured evaluation loss for SEA-RAFT on a fixed pair went 0.404 (untrained), 38.6, 100.2, 1.26, ... , 0.111 by step 15. Comparing step 1 against step 2 read 38.6 against 100.2 and called a model that ends up 3.6x BETTER than untrained a regression.
>
> ... It is **transient-immune by construction, so it needs no per-model knowledge of where a given architecture stops oscillating**.

`NERModelTestBase` overrides that method and keeps the old short-vs-long comparison. The override exists for a good and still-valid reason — NER `Predict` argmax-decodes to discrete label IDs, so the base's `MeasureLoss(net, net.Predict(input), target)` is meaningless for NER and it uses `GetLastLoss()` instead — but it carried the superseded *comparison* along with the fix it actually needed.

## The per-model tolerance is the smell, and it is already stale

`TestScaffoldGenerator` carries a LegalBERTNER-specific tolerance:

```csharp
// ... its second Adam step rises further than the shared 0.5 bound allows: measured
// GetLastLoss 2.847519 after one step against 3.455980 after two, a gap of 0.608.
double moreDataTolerance = model.ClassName == "LegalBERTNER" ? 1.0 : 0.5;
```

That is precisely the "per-model knowledge of where a given architecture stops oscillating" the base says the invariant should not need. And it no longer holds: the gap measured when it was written was **0.608**, the nightly measured **1.167**. Widening it again would be chasing a number that moves with initialisation and dropout, not fixing anything.

## Why this is a test defect rather than a model defect

Nothing else suggests LegalBERTNER is broken:

- `Training_ShouldReduceLoss` passes
- `Training_ShouldChangeParameters` passes
- `TrainingError_ShouldNotExceedTestError`, `GradientFlow_ShouldBeNonZeroAndFinite`, `Gradients_MatchFiniteDifference` all pass in the same nightly run
- `TransformerNEROptions.LearningRate` defaults to `5e-5`, which is the BERT fine-tuning rate its paper (Chalkidis et al., EMNLP 2020 Findings) uses — not an obviously wrong rate
- `SequenceLabelingNERBase` pairs correctly with `CrossEntropyWithLogitsLoss`, so this is not the raw-logit/`CategoricalCrossEntropy` mispairing fixed for the LM heads

A BERT encoder two AdamW steps from initialisation is squarely inside the transient the base comment describes.

## Suggested fix

Bring the NER override in line with the base's semantics — keep `GetLastLoss()` (the reason the override exists), drop the short-vs-long comparison and the per-model tolerance in favour of an adequate budget measured against a pre-training reference.

Also worth noting: `TransformerNERBase.CreateDefaultOptimizer()` only attaches its `LinearWarmupScheduler` when `WarmupSteps > 0 || TotalTrainingSteps > 0`, so by default these models take raw AdamW steps from initialisation with no warmup — the condition BERT-family fine-tuning normally uses warmup to avoid. Whether the default should include warmup is a separate question from the test invariant, and should not be settled by making a test pass.

Related: #2087 (the lane this hid behind), #2134 (the OOM fix), #1679 (the earlier NER override rationale).
